### PR TITLE
feat: add pin and unpin methods

### PR DESCRIPTION
**Pull request by rrrrs09**
# Description

Added pin and unpin methods

# Checklist

- [x] Linters and tests have passed
- [x] I've written a changelog for PR change
- [x] I'll make a release when PR is approved

### CODE REVIEW

**Review by lxmnk — APPROVED** [review]

File: botx/bots/mixins/requests/chats.py [hunk 1]
```diff
@@ -188,3 +190,37 @@ async def add_admin_roles(
             AddAdminRole(group_chat_id=chat_id, user_huids=user_huids),
             credentials=credentials,
         )
+
+    async def pin_message(
+        self: BotXMethodCallProtocol,
+        credentials: SendingCredentials,
+        chat_id: UUID,
+        sync_id: UUID,
```
Comment (Opti213) [inline botx/bots/mixins/requests/chats.py:198]: credentials has chat(and message) id, maybe use them as default?

File: botx/clients/methods/errors/no_permission.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+"""Definition for "no permission" error."""
```
Comment (Opti213) [inline botx/clients/methods/errors/no_permission.py:1]: `permissions` as name of module would be better, i think

File: botx/clients/methods/errors/no_permission.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+
+class NoPermissionErrorData(BaseModel):
+    """Data for error when there is no permission for operation."""
+
+    #: ID of chat that was requested.
+    group_chat_id: UUID
+
+
+def handle_error(method: BotXMethod, response: HTTPResponse) -> NoReturn:
```
Comment (Opti213) [inline botx/clients/methods/errors/no_permission.py:30]: You use `None` as annotation in all other places, and `None` as no-return annotation used in all project

File: botx/bots/mixins/requests/chats.py (hunk 1 above)
Comment (rrrrs09) [reply botx/bots/mixins/requests/chats.py:198]: Custom credentials can be passed which may not have chat and sync IDs as these fields are optional.

File: botx/clients/methods/errors/no_permission.py (hunk 3 above)
Comment (rrrrs09) [reply botx/clients/methods/errors/no_permission.py]: NoReturn is used in all modules with errors in  botx.clients.methods.errors

File: botx/clients/methods/errors/no_permission.py (hunk 2 above)
Comment (rrrrs09) [reply botx/clients/methods/errors/no_permission.py]: done

**Review by Opti213 — APPROVED** [review]

**Pull request merged**